src/drivers/driver_zephyr.c

```diff
@@ -1812,7 +1832,11 @@ static int wpa_drv_zep_set_supp_port(void *priv,
 
 #ifdef CONFIG_NET_DHCPV4
 	if (authorized) {
-		net_dhcpv4_restart(iface);
+		if (if_ctx->ft_roaming == false) {
+			net_dhcpv4_restart(iface);
+		} else {
+			if_ctx->ft_roaming = false;
+		}
```

Comment:
This only works for L2 roaming, should this be a Kconfig option? or an alternative solution?

Reply:
What do you mean by Kconfig option?

Reply:
Something like below? Also please extend the commit log to explain

```
config WIFI_NM_WPA_SUPPLICANT_SKIP_DHCP_ON_ROAMING
   bool "Do not restart DHCP when roaming e.g., L2 roaming"
   help
     For L2 roaming, the IP network doesn't change, so, DHCP can be skipped.
     For L3 roaming this needs to be enabled.
```

Reply:
Could you please help clarify L2 roaming and L3 roaming? Thanks!

Reply:
L2 romaing, romaing within same subnet, L3 roaming is roaming across subnets, some info https://learningnetwork.cisco.com/s/question/0D56e0000CjUe4lCQC/what-is-layer2-and-layer3-roaming . The point is this decision of whether to perform DHCP renewal or not should be user configurable as each scenario might need a different behaviour.

Reply:
Ok, got it. I will update the PR later. Thanks!

Reply:
Hi @krish2718 , I added new kconfig WIFI_NM_WPA_SUPPLICANT_SKIP_DHCP_ON_ROAMING as suggested and wrapped the source code in this new config. Please help review it. Thanks!